src/Microsoft.VisualStudio.ProjectSystem.Managed.VS/ProjectSystem/VS/Debug/ProjectDebuggerProvider.cs

```diff
@@ -0,0 +1,338 @@
+        public IDebugProfileLaunchTargetsProvider LastLaunchProvider { get; private set; }
+
+        /// <summary>
+        /// Called by CPS to determine whether we can launch
+        /// </summary>
+        public override Task<bool> CanLaunchAsync(DebugLaunchOptions launchOptions)
+        {
+            return TplExtensions.TrueTask;
+        }
+
+        /// <summary>
+        /// Helper returns the correct debugger engine based on the targeted framework
+        /// </summary>
+        public static Guid GetDebugEngineForFramework(string targetFramework)
+        {
+            // The engine depends on the framework
+            if (IsDotNetCoreFramework(targetFramework))
+            {
+                return  DebuggerEngines.ManagedCoreEngine;
+            }
+            else
+            {
+                return DebuggerEngines.ManagedOnlyEngine;
+            }
+        }
+
+        /// <summary>
+        /// This is a placeholder
+        /// </summary>
+        public static bool IsDotNetCoreFramework( string targetFramework)
+        {
+            const string NetStandardAppPrefix = ".NetStandardApp";
+            const string NetStandardPrefix = ".NetStandard";
+            const string NetCorePrefix = ".NetCore";
+            const string NetCoreAppPrefix = ".NetCoreApp";
+            return targetFramework.StartsWith(NetCoreAppPrefix, StringComparison.OrdinalIgnoreCase) ||
+                   targetFramework.StartsWith(NetCorePrefix, StringComparison.OrdinalIgnoreCase) ||
+                   targetFramework.StartsWith(NetStandardPrefix, StringComparison.OrdinalIgnoreCase) ||
+                   targetFramework.StartsWith(NetStandardAppPrefix, StringComparison.OrdinalIgnoreCase);
+        }
+
+        /// <summary>
+        /// This is called on F5 to return the list of debug targets.What we return depends on the type
+        /// of project.  
+        /// </summary>
+        public override async Task<IReadOnlyList<IDebugLaunchSettings>> QueryDebugTargetsAsync(DebugLaunchOptions launchOptions)
+        {
+            // Get the active debug profile
+            ILaunchProfile activeProfile = LaunchSettingsProvider.ActiveProfile;
+
+            // Should have a profile
+            if(activeProfile == null)
+            {
+                throw new Exception(VSResources.ActiveLaunchProfileNotFound);
+            }
+
+            // Now find the DebugTargets provider for this profile
+            var launchProvider = GetLaunchTargetsProviderForProfile(activeProfile);
+            if(launchProvider == null)
+            {
+                throw new Exception(string.Format(VSResources.DontKnowHowToRunProfile, activeProfile.Name));
+            }
+
+            var launchSettings = await launchProvider.QueryDebugTargetsAsync(launchOptions, activeProfile).ConfigureAwait(true);
+            LastLaunchProvider = launchProvider;
```

Comment:
I'm pretty sure it's not, but to just to be sure, is a race possible here? ie, a call to an `IDeployedProjectMappingProvider` gets the wrong launch provider returned?